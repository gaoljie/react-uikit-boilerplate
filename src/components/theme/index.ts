import React from "react";

export const theme = {
  primary: "#EF6F35",
  primaryFont: "Montserrat",
  secondaryFont: "Work Sans",
  desktop: "@media (max-width: 1920px)",
  smallDesktop: "@media (max-width: 1441px)",
  tablet: "@media (max-width: 769px)",
  mobile: "@media (max-width: 600px)"
};

export const ThemeContext = React.createContext(theme);

export const ThemeProvider = ThemeContext.Provider;
