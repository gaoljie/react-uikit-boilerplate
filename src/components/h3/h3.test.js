import React from "react";
import "jest-styled-components";
import { mount } from "enzyme";
import theme from "../theme";
import H3 from "./index";

describe("Test H3", () => {
  it("H3 Test", () => {
    const wrapper = mount(<H3>Primary</H3>);

    expect(wrapper).toMatchSnapshot();

    expect(wrapper).toHaveStyleRule("color", "#333333");

    expect(wrapper).toHaveStyleRule("font-family", theme.primaryFont);

    expect(wrapper.children().text()).toEqual("Primary");

    expect(wrapper.exists("h3")).toEqual(true);
  });
});
