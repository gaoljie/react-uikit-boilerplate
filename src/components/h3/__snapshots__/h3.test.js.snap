// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test H3 H3 Test 1`] = `
.c0 {
  font-family: Montserrat;
  color: #333333;
  font-size: 22px;
  line-height: 30px;
}

@media (max-width:769px) {
  .c0 {
    font-size: 18px;
    line-height: 26px;
  }
}

<H3>
  <Styled(styled.div)
    as="h3"
  >
    <StyledComponent
      as="h3"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "lastClassName": "c0",
            "rules": Array [
              "
  ",
              [Function],
              "
",
              "
  font-family: ",
              "Montserrat",
              ";
  color: ",
              [Function],
              ";
  font-size: 22px;
  line-height: 30px;

  ",
              "@media (max-width: 769px)",
              " {
    font-size: 18px;
    line-height: 26px;
  }
",
            ],
          },
          "displayName": "Styled(styled.div)",
          "foldedComponentIds": Array [
            "sc-bdVaJa",
          ],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <h3
        className="c0"
      >
        Primary
      </h3>
    </StyledComponent>
  </Styled(styled.div)>
</H3>
`;
